sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/compatibility/AzureSpringBootVersionVerifier.java

```diff
@@ -68,19 +93,23 @@ private boolean springBootVersionMatches() {
                     return true;
                 }
             } catch (FileNotFoundException e) {
-                CompatibilityPredicate predicate = this.supportedVersions.get(stripWildCardFromVersion(acceptedVersion));
-                if (predicate != null && predicate.isCompatible()) {
+                String versionString = stripWildCardFromVersion(acceptedVersion);
+                String fullyQuallifiedClassName = this.supportedVersions.get(versionString);
+
+                if (test(fullyQuallifiedClassName)) {
                     if (LOGGER.isDebugEnabled()) {
-                        LOGGER.debug("Predicate [" + predicate + "] was matched");
+                        LOGGER.debug("Predicate [" + versionString + "] was matched");
```

Comment:
`predicate` is not a version string,  `Predicate` followed by `versionString` is strange.

Reply:
update this log